python/opentrons_ot3_firmware/messages/payloads.py

```diff
@@ -149,3 +149,54 @@ class ReadPresenceSensingVoltageResponsePayload(utils.BinarySerializable):
     z_motor: utils.UInt16Field
     a_motor: utils.UInt16Field
     gripper: utils.UInt16Field
+
+
+@dataclass
+class FirmwareUpdateWithAddress(utils.BinarySerializable):
+    """A FW update payload with an address."""
+
+    address: utils.UInt32Field
+
+
+class FirmwareUpdateDataField(utils.BinaryFieldBase[bytes]):
+    """The data field of FirmwareUpdateData."""
+    NUM_BYTES = 56
+    FORMAT = f"{NUM_BYTES}s"
+
+
+@dataclass
+class FirmwareUpdateData(FirmwareUpdateWithAddress):
+    """A FW update data payload."""
+
+    num_bytes: utils.UInt8Field
+    reserved: utils.UInt8Field
+    data: FirmwareUpdateDataField
```

Comment:
Data is a fixed 56 bytes. `num_bytes` indicates how many of the 56 are to be used on the off chance that not all of them are used.